pkg/util/winutil/pdhutil/pdhcounter.go

```diff
@@ -37,306 +34,137 @@ type CounterInstanceVerify func(string) bool
-	/*
-	   When handling non english versions, the counters don't work quite as documented.
-	   This is because strings like "Bytes Sent/sec" might appear multiple times in the
-	   english master, and might not have mappings for each index.
-
-	   Search each index, and make sure the requested counter name actually appears in
-	   the list of available counters; that's the counter we'll use.
-
-	   For more information, see README.md.
-	*/
-	idxList, err := getCounterIndexList(counterName)
-	if err != nil {
-		return "", err
-	}
-	for _, ndx := range idxList {
-		counter, e := pfnPdhLookupPerfNameByIndex(ndx)
-		if e != nil {
-			log.Debugf("Counter index %d not found, skipping", ndx)
-			continue
-		}
-		// see if the counter we got back is in the list of counters
-		if !stringInSlice(counter, counters) {
-			log.Debugf("counter %s not in counter list", counter)
-			continue
-		}
-		// check to see if we can create the counter
-		path, err := pfnPdhMakeCounterPath(machine, p.className, instanceName, counter)
-		if err == nil {
-			log.Debugf("Successfully created counter path %s", path)
-			p.counterName = counter
-			return path, nil
-		}
-		// else
-		log.Debugf("Unable to create path with %s, trying again", counter)
-	}
-	// if we get here, was never able to find a counter path or create a valid
-	// path.  Return failure.
-	log.Warnf("Unable to create counter path for %s %s", counterName, instanceName)
-	return "", fmt.Errorf("Unable to create counter path %s %s", counterName, instanceName)
-}
-
-// GetAllValues returns the data associated with each instance in a query.
-func (p *PdhMultiInstanceCounterSet) GetAllValues() (values map[string]float64, err error) {
-	values = make(map[string]float64)
-	err = nil
-	var removeList []string
-	pfnPdhCollectQueryData(p.query)
-	for inst, hcounter := range p.countermap {
-		var retval float64
-		retval, err = pfnPdhGetFormattedCounterValueFloat(hcounter)
-		if err != nil {
-			switch err.(type) {
-			case *ErrPdhInvalidInstance:
-				removeList = append(removeList, inst)
-				log.Debugf("Got invalid instance for %s %s", p.requestedCounterName, inst)
-				err = nil
-				continue
-			default:
-				log.Debugf("Other Error getting all values %s %s %v", p.requestedCounterName, inst, err)
-				return
-			}
-		}
-		values[inst] = retval
-	}
-	for _, inst := range removeList {
-		p.RemoveInvalidInstance(inst)
-	}
-	// check for newly found instances
-	p.MakeInstanceList()
-	return
+	return values, nil
 }
 
 // GetValue returns the data associated with a single-value counter
 func (p *PdhSingleInstanceCounterSet) GetValue() (val float64, err error) {
-	if p.singleCounter == PDH_HCOUNTER(0) {
+	if p.counter == PDH_HCOUNTER(0) {
 		return 0, fmt.Errorf("Not a single-value counter")
 	}
 	pfnPdhCollectQueryData(p.query)
-	return pfnPdhGetFormattedCounterValueFloat(p.singleCounter)
+	return pfnPdhGetFormattedCounterValueFloat(p.counter)
```

Comment:
Interesting, I just realized. Do we create an individual query per counter? That seems to be bit excessive hence single call for multiple counters would work probably better.

Reply:
> Do we create an individual query per counter?

That's correct. The # of counters is small though, four in `memory_windows.go` and one in `file_handles_windows.go` for example.

What would you suggest? A single global query? Each check opens a query, then adds counters to it?

Reply:
Am I wrong with 19 in the lower comment? I feels that single global query would be better however considering that checks are semi-independent that is probably harder to arrange and can make a code more brittle. At least we can have a single per check. No need to do anything now but something consider in the future.